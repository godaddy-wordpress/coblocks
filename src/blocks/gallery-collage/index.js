/**
 * External dependencies
 */
import { GalleryCollageIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * WordPress dependencies.
 */
import { __ } from '@wordpress/i18n';
import { Icon, Spinner } from '@wordpress/components';
import { lazy, Suspense } from '@wordpress/element';

/**
 * Internal dependencies.
 */
import deprecated from './deprecated';
const Edit = lazy( () => import( './edit' ) );
import { GalleryAttributes } from '../../components/block-gallery/shared';
import { hasFormattingCategory } from '../../utils/block-helpers';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';

/**
 * Block constants.
 */
const { name, category } = metadata;

const attributes = {
	...GalleryAttributes,
	...metadata.attributes,
	gutter: {
		type: 'string',
		default: 'small',
	},
};

const settings = {
	category: hasFormattingCategory ? 'coblocks-galleries' : 'media',
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'gallery', 'coblocks' ),
		/* translators: block keyword */
		__( 'photos', 'coblocks' ),
	],
	styles: [
		{
			name: 'default',
			/* translators: block style */
			label: __( 'Default', 'coblocks' ),
			isDefault: true,
		},
		{
			name: 'tiled',
			/* translators: block style */
			label: __( 'Tiled', 'coblocks' ),
		},
		{
			name: 'layered',
			/* translators: block style */
			label: __( 'Layered', 'coblocks' ),
		},
	],
	example: {
		attributes: {
			images: [
				{ index: 0, url: 'https://s.w.org/images/core/5.3/Sediment_off_the_Yucatan_Peninsula.jpg' },
				{ index: 1, url: 'https://s.w.org/images/core/5.3/Windbuchencom.jpg' },
				{ index: 2, url: 'https://s.w.org/images/core/5.3/Biologia_Centrali-Americana_-_Cantorchilus_semibadius_1902.jpg' },
				{ index: 3, url: 'https://s.w.org/images/core/5.3/Glacial_lakes,_Bhutan.jpg' },
				{ index: 4, url: 'https://s.w.org/images/core/5.3/MtBlanc1.jpg' },
			],
		},
	},
	attributes,
	transforms,
	edit: ( props ) => (
		<Suspense fallback={ <Spinner /> }>
			<Edit { ...props } />
		</Suspense>
	),
	save,
	deprecated,
};

export { name, category, metadata, settings };
