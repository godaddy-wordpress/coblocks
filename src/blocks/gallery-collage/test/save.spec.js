/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { createBlock, serialize } from '@wordpress/blocks';

/**
 * Internal dependencies.
 */
import * as settings from '../index';
import { registerBlock } from '../../../../src/utils/helper';

// Make variables accessible for all tests.
let block;
let serializedBlock;

const baseAttributes = {
	images: [
		{ url: 'https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg', id: 1, index: 0 },
		{ url: 'https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg', id: 2, index: 1 },
	],
};
const name = settings?.metadata?.name ?? settings?.name;

describe( 'coblocks/gallery-collage', () => {
	beforeAll( () => {
		// Register the block.
		registerBlock( settings );
	} );

	beforeEach( () => {
		// Create the block with the minimum attributes.
		block = createBlock( name, baseAttributes );
		// Reset the reused variables.
		serializedBlock = '';
	} );

	it( 'should render with images', () => {
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( baseAttributes.images[ 0 ].url );
		expect( serializedBlock ).toContain( `data-id="${ baseAttributes.images[ 0 ].id }"` );
		expect( serializedBlock ).toContain( `wp-image-${ baseAttributes.images[ 0 ].id }` );

		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should not render image if URL is not defined', () => {
		block.attributes = { ...block.attributes, images: [ { id: 1, index: 0 } ] };
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).not.toContain( 'img' );

		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should have className \'has-lightbox\' with lightbox enabled.', () => {
		block.attributes = { ...block.attributes, lightbox: true };
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'has-lightbox' );

		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should have figcaption element with captions enabled and caption text.', () => {
		block.attributes = {
			...block.attributes,
			images: [
				...block.attributes.images,
				{ url: 'https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg', id: 3, index: 3, caption: 'test caption' },
			],
			captions: false,
		};

		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).not.toContain( '<figcaption class="wp-block-coblocks-gallery-collage__caption">' );

		block.attributes.captions = true;
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '<figcaption class="wp-block-coblocks-gallery-collage__caption">' );

		expect( serializedBlock ).toMatchSnapshot();
	} );

	[ 'light', 'dark' ].forEach( ( captionStyle ) => {
		it( `should have className \'has-caption-style-${ captionStyle }\' with captionStyle set to '${ captionStyle }'.`, () => {
			block.attributes = {
				...block.attributes,
				images: [
					...block.attributes.images,
					{ url: 'https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg', id: 3, index: 3, caption: 'test caption' },
				],
				captions: true,
				captionStyle,
			};

			serializedBlock = serialize( block );

			expect( serializedBlock ).toBeDefined();
			expect( serializedBlock ).toContain( `has-caption-style-${ captionStyle }` );

			expect( serializedBlock ).toMatchSnapshot();
		} );
	} );

	[ 'is-style-tiled', 'is-style-layered' ].forEach( ( layoutStyle ) => {
		it( `should have className \'${ layoutStyle }\'.`, () => {
			block.attributes = {
				...block.attributes,
				className: layoutStyle,
			};

			serializedBlock = serialize( block );

			expect( serializedBlock ).toBeDefined();
			expect( serializedBlock ).toContain( layoutStyle );

			expect( serializedBlock ).toMatchSnapshot();
		} );
	} );

	it( 'should have custom link when linkTo is set to "custom"', () => {
		block.attributes = {
			...block.attributes,
			images: [
				...block.attributes.images,
				{ url: 'https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg', id: 3, index: 3, imgLink: 'http://google.com' },
			],
			linkTo: 'custom',
		};

		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'href="http://google.com"' );

		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should have media link when linkTo is set to "media"', () => {
		block.attributes = {
			...block.attributes,
			images: [
				...block.attributes.images,
				{ url: 'https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg', id: 3, index: 3 },
			],
			linkTo: 'media',
		};

		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'href="https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg"' );

		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should have attachment link when linkTo is set to "attachment"', () => {
		block.attributes = {
			...block.attributes,
			images: [
				...block.attributes.images,
				{ url: 'https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg', id: 3, index: 3, link: 'https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg' },
			],
			linkTo: 'attachment',
		};

		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'href="https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg"' );

		expect( serializedBlock ).toMatchSnapshot();
	} );

	[ 'grayscale', 'sepia', 'saturation', 'dim', 'vintage' ].forEach( ( filter ) => {
		it( `should have className \'has-filter-${ filter }\' with filter set to '${ filter }'.`, () => {
			block.attributes = { ...block.attributes, filter };
			serializedBlock = serialize( block );

			expect( serializedBlock ).toBeDefined();
			expect( serializedBlock ).toContain( `has-filter-${ filter }` );

			expect( serializedBlock ).toMatchSnapshot();
		} );
	} );

	[ 'small', 'medium', 'large' ].forEach( ( gutter ) => {
		it( `should have className 'has-${ gutter }-gutter' with gutter set to '${ gutter }'.`, () => {
			block.attributes = { ...block.attributes, gutter };

			serializedBlock = serialize( block );

			expect( serializedBlock ).toBeDefined();
			expect( serializedBlock ).toContain( `has-${ gutter }-gutter` );

			expect( serializedBlock ).toMatchSnapshot();
		} );
	} );
} );
