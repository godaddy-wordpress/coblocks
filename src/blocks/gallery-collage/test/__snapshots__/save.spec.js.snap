// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coblocks/gallery-collage should have attachment link when linkTo is set to "attachment" 1`] = `
"<!-- wp:coblocks/gallery-collage {\\"linkTo\\":\\"attachment\\"} -->
<div aria-label=\\"Collage Gallery\\" class=\\"wp-block-coblocks-gallery-collage alignwide has-small-gutter\\"><ul><li class=\\"wp-block-coblocks-gallery-collage__item item-1\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-1\\" data-id=\\"1\\" data-index=\\"0\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"wp-block-coblocks-gallery-collage__item item-2\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-2\\" data-id=\\"2\\" data-index=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li><li class=\\"wp-block-coblocks-gallery-collage__item item-3\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><a href=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\" rel=\\"\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-3\\" data-id=\\"3\\" data-index=\\"3\\" data-link=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/></a></figure></li></ul></div>
<!-- /wp:coblocks/gallery-collage -->"
`;

exports[`coblocks/gallery-collage should have className 'has-caption-style-dark' with captionStyle set to 'dark'. 1`] = `
"<!-- wp:coblocks/gallery-collage {\\"captions\\":true} -->
<div aria-label=\\"Collage Gallery\\" class=\\"wp-block-coblocks-gallery-collage alignwide has-caption-style-dark has-small-gutter\\"><ul><li class=\\"wp-block-coblocks-gallery-collage__item item-1\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-1\\" data-id=\\"1\\" data-index=\\"0\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"wp-block-coblocks-gallery-collage__item item-2\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-2\\" data-id=\\"2\\" data-index=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li><li class=\\"wp-block-coblocks-gallery-collage__item item-3\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-3\\" data-id=\\"3\\" data-index=\\"3\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/><figcaption class=\\"wp-block-coblocks-gallery-collage__caption\\">test caption</figcaption></figure></li></ul></div>
<!-- /wp:coblocks/gallery-collage -->"
`;

exports[`coblocks/gallery-collage should have className 'has-caption-style-light' with captionStyle set to 'light'. 1`] = `
"<!-- wp:coblocks/gallery-collage {\\"captionStyle\\":\\"light\\",\\"captions\\":true} -->
<div aria-label=\\"Collage Gallery\\" class=\\"wp-block-coblocks-gallery-collage alignwide has-caption-style-light has-small-gutter\\"><ul><li class=\\"wp-block-coblocks-gallery-collage__item item-1\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-1\\" data-id=\\"1\\" data-index=\\"0\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"wp-block-coblocks-gallery-collage__item item-2\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-2\\" data-id=\\"2\\" data-index=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li><li class=\\"wp-block-coblocks-gallery-collage__item item-3\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-3\\" data-id=\\"3\\" data-index=\\"3\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/><figcaption class=\\"wp-block-coblocks-gallery-collage__caption\\">test caption</figcaption></figure></li></ul></div>
<!-- /wp:coblocks/gallery-collage -->"
`;

exports[`coblocks/gallery-collage should have className 'has-filter-dim' with filter set to 'dim'. 1`] = `
"<!-- wp:coblocks/gallery-collage {\\"filter\\":\\"dim\\"} -->
<div aria-label=\\"Collage Gallery\\" class=\\"wp-block-coblocks-gallery-collage alignwide has-filter-dim has-small-gutter\\"><ul><li class=\\"wp-block-coblocks-gallery-collage__item item-1\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-1\\" data-id=\\"1\\" data-index=\\"0\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"wp-block-coblocks-gallery-collage__item item-2\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-2\\" data-id=\\"2\\" data-index=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li></ul></div>
<!-- /wp:coblocks/gallery-collage -->"
`;

exports[`coblocks/gallery-collage should have className 'has-filter-grayscale' with filter set to 'grayscale'. 1`] = `
"<!-- wp:coblocks/gallery-collage {\\"filter\\":\\"grayscale\\"} -->
<div aria-label=\\"Collage Gallery\\" class=\\"wp-block-coblocks-gallery-collage alignwide has-filter-grayscale has-small-gutter\\"><ul><li class=\\"wp-block-coblocks-gallery-collage__item item-1\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-1\\" data-id=\\"1\\" data-index=\\"0\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"wp-block-coblocks-gallery-collage__item item-2\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-2\\" data-id=\\"2\\" data-index=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li></ul></div>
<!-- /wp:coblocks/gallery-collage -->"
`;

exports[`coblocks/gallery-collage should have className 'has-filter-saturation' with filter set to 'saturation'. 1`] = `
"<!-- wp:coblocks/gallery-collage {\\"filter\\":\\"saturation\\"} -->
<div aria-label=\\"Collage Gallery\\" class=\\"wp-block-coblocks-gallery-collage alignwide has-filter-saturation has-small-gutter\\"><ul><li class=\\"wp-block-coblocks-gallery-collage__item item-1\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-1\\" data-id=\\"1\\" data-index=\\"0\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"wp-block-coblocks-gallery-collage__item item-2\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-2\\" data-id=\\"2\\" data-index=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li></ul></div>
<!-- /wp:coblocks/gallery-collage -->"
`;

exports[`coblocks/gallery-collage should have className 'has-filter-sepia' with filter set to 'sepia'. 1`] = `
"<!-- wp:coblocks/gallery-collage {\\"filter\\":\\"sepia\\"} -->
<div aria-label=\\"Collage Gallery\\" class=\\"wp-block-coblocks-gallery-collage alignwide has-filter-sepia has-small-gutter\\"><ul><li class=\\"wp-block-coblocks-gallery-collage__item item-1\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-1\\" data-id=\\"1\\" data-index=\\"0\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"wp-block-coblocks-gallery-collage__item item-2\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-2\\" data-id=\\"2\\" data-index=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li></ul></div>
<!-- /wp:coblocks/gallery-collage -->"
`;

exports[`coblocks/gallery-collage should have className 'has-filter-vintage' with filter set to 'vintage'. 1`] = `
"<!-- wp:coblocks/gallery-collage {\\"filter\\":\\"vintage\\"} -->
<div aria-label=\\"Collage Gallery\\" class=\\"wp-block-coblocks-gallery-collage alignwide has-filter-vintage has-small-gutter\\"><ul><li class=\\"wp-block-coblocks-gallery-collage__item item-1\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-1\\" data-id=\\"1\\" data-index=\\"0\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"wp-block-coblocks-gallery-collage__item item-2\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-2\\" data-id=\\"2\\" data-index=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li></ul></div>
<!-- /wp:coblocks/gallery-collage -->"
`;

exports[`coblocks/gallery-collage should have className 'has-large-gutter' with gutter set to 'large'. 1`] = `
"<!-- wp:coblocks/gallery-collage {\\"gutter\\":\\"large\\"} -->
<div aria-label=\\"Collage Gallery\\" class=\\"wp-block-coblocks-gallery-collage alignwide has-large-gutter\\"><ul><li class=\\"wp-block-coblocks-gallery-collage__item item-1\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-1\\" data-id=\\"1\\" data-index=\\"0\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"wp-block-coblocks-gallery-collage__item item-2\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-2\\" data-id=\\"2\\" data-index=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li></ul></div>
<!-- /wp:coblocks/gallery-collage -->"
`;

exports[`coblocks/gallery-collage should have className 'has-lightbox' with lightbox enabled. 1`] = `
"<!-- wp:coblocks/gallery-collage {\\"lightbox\\":true} -->
<div aria-label=\\"Collage Gallery\\" class=\\"wp-block-coblocks-gallery-collage alignwide has-lightbox has-small-gutter\\"><ul><li class=\\"wp-block-coblocks-gallery-collage__item item-1\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-1\\" data-id=\\"1\\" data-index=\\"0\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"wp-block-coblocks-gallery-collage__item item-2\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-2\\" data-id=\\"2\\" data-index=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li></ul></div>
<!-- /wp:coblocks/gallery-collage -->"
`;

exports[`coblocks/gallery-collage should have className 'has-medium-gutter' with gutter set to 'medium'. 1`] = `
"<!-- wp:coblocks/gallery-collage {\\"gutter\\":\\"medium\\"} -->
<div aria-label=\\"Collage Gallery\\" class=\\"wp-block-coblocks-gallery-collage alignwide has-medium-gutter\\"><ul><li class=\\"wp-block-coblocks-gallery-collage__item item-1\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-1\\" data-id=\\"1\\" data-index=\\"0\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"wp-block-coblocks-gallery-collage__item item-2\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-2\\" data-id=\\"2\\" data-index=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li></ul></div>
<!-- /wp:coblocks/gallery-collage -->"
`;

exports[`coblocks/gallery-collage should have className 'has-small-gutter' with gutter set to 'small'. 1`] = `
"<!-- wp:coblocks/gallery-collage -->
<div aria-label=\\"Collage Gallery\\" class=\\"wp-block-coblocks-gallery-collage alignwide has-small-gutter\\"><ul><li class=\\"wp-block-coblocks-gallery-collage__item item-1\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-1\\" data-id=\\"1\\" data-index=\\"0\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"wp-block-coblocks-gallery-collage__item item-2\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-2\\" data-id=\\"2\\" data-index=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li></ul></div>
<!-- /wp:coblocks/gallery-collage -->"
`;

exports[`coblocks/gallery-collage should have className 'is-style-layered'. 1`] = `
"<!-- wp:coblocks/gallery-collage {\\"className\\":\\"is-style-layered\\"} -->
<div aria-label=\\"Collage Gallery\\" class=\\"wp-block-coblocks-gallery-collage alignwide is-style-layered has-small-gutter\\"><ul><li class=\\"wp-block-coblocks-gallery-collage__item item-1\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-1\\" data-id=\\"1\\" data-index=\\"0\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"wp-block-coblocks-gallery-collage__item item-2\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-2\\" data-id=\\"2\\" data-index=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li></ul></div>
<!-- /wp:coblocks/gallery-collage -->"
`;

exports[`coblocks/gallery-collage should have className 'is-style-tiled'. 1`] = `
"<!-- wp:coblocks/gallery-collage {\\"className\\":\\"is-style-tiled\\"} -->
<div aria-label=\\"Collage Gallery\\" class=\\"wp-block-coblocks-gallery-collage alignwide is-style-tiled has-small-gutter\\"><ul><li class=\\"wp-block-coblocks-gallery-collage__item item-1\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-1\\" data-id=\\"1\\" data-index=\\"0\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"wp-block-coblocks-gallery-collage__item item-2\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-2\\" data-id=\\"2\\" data-index=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li></ul></div>
<!-- /wp:coblocks/gallery-collage -->"
`;

exports[`coblocks/gallery-collage should have custom link when linkTo is set to "custom" 1`] = `
"<!-- wp:coblocks/gallery-collage {\\"linkTo\\":\\"custom\\"} -->
<div aria-label=\\"Collage Gallery\\" class=\\"wp-block-coblocks-gallery-collage alignwide has-small-gutter\\"><ul><li class=\\"wp-block-coblocks-gallery-collage__item item-1\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-1\\" data-id=\\"1\\" data-index=\\"0\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"wp-block-coblocks-gallery-collage__item item-2\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-2\\" data-id=\\"2\\" data-index=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li><li class=\\"wp-block-coblocks-gallery-collage__item item-3\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><a href=\\"http://google.com\\" rel=\\"\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-3\\" data-id=\\"3\\" data-imglink=\\"http://google.com\\" data-index=\\"3\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/></a></figure></li></ul></div>
<!-- /wp:coblocks/gallery-collage -->"
`;

exports[`coblocks/gallery-collage should have figcaption element with captions enabled and caption text. 1`] = `
"<!-- wp:coblocks/gallery-collage {\\"captions\\":true} -->
<div aria-label=\\"Collage Gallery\\" class=\\"wp-block-coblocks-gallery-collage alignwide has-caption-style-dark has-small-gutter\\"><ul><li class=\\"wp-block-coblocks-gallery-collage__item item-1\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-1\\" data-id=\\"1\\" data-index=\\"0\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"wp-block-coblocks-gallery-collage__item item-2\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-2\\" data-id=\\"2\\" data-index=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li><li class=\\"wp-block-coblocks-gallery-collage__item item-3\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-3\\" data-id=\\"3\\" data-index=\\"3\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/><figcaption class=\\"wp-block-coblocks-gallery-collage__caption\\">test caption</figcaption></figure></li></ul></div>
<!-- /wp:coblocks/gallery-collage -->"
`;

exports[`coblocks/gallery-collage should have media link when linkTo is set to "media" 1`] = `
"<!-- wp:coblocks/gallery-collage {\\"linkTo\\":\\"media\\"} -->
<div aria-label=\\"Collage Gallery\\" class=\\"wp-block-coblocks-gallery-collage alignwide has-small-gutter\\"><ul><li class=\\"wp-block-coblocks-gallery-collage__item item-1\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><a href=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\" rel=\\"\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-1\\" data-id=\\"1\\" data-index=\\"0\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></a></figure></li><li class=\\"wp-block-coblocks-gallery-collage__item item-2\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><a href=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\" rel=\\"\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-2\\" data-id=\\"2\\" data-index=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></a></figure></li><li class=\\"wp-block-coblocks-gallery-collage__item item-3\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><a href=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\" rel=\\"\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-3\\" data-id=\\"3\\" data-index=\\"3\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-3.jpg\\"/></a></figure></li></ul></div>
<!-- /wp:coblocks/gallery-collage -->"
`;

exports[`coblocks/gallery-collage should not render image if URL is not defined 1`] = `
"<!-- wp:coblocks/gallery-collage -->
<div aria-label=\\"Collage Gallery\\" class=\\"wp-block-coblocks-gallery-collage alignwide has-small-gutter\\"><ul><li class=\\"wp-block-coblocks-gallery-collage__item item-1\\"></li></ul></div>
<!-- /wp:coblocks/gallery-collage -->"
`;

exports[`coblocks/gallery-collage should render with images 1`] = `
"<!-- wp:coblocks/gallery-collage -->
<div aria-label=\\"Collage Gallery\\" class=\\"wp-block-coblocks-gallery-collage alignwide has-small-gutter\\"><ul><li class=\\"wp-block-coblocks-gallery-collage__item item-1\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-1\\" data-id=\\"1\\" data-index=\\"0\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-1.jpg\\"/></figure></li><li class=\\"wp-block-coblocks-gallery-collage__item item-2\\"><figure class=\\"wp-block-coblocks-gallery-collage__figure\\"><img alt=\\"Gallery Image\\" class=\\"wp-image-2\\" data-id=\\"2\\" data-index=\\"1\\" src=\\"https://wordpress.com/wp-content/uploads/1234/56/image-2.jpg\\"/></figure></li></ul></div>
<!-- /wp:coblocks/gallery-collage -->"
`;
